### ISSUE TITLE
Fix comma-dangle schema

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#13739]
`comma-dangle` schema should be fixed to allow only 1 element (either string or object, not string + object) in the configuration array, since the rule reads only the first element and ignores the rest:

> This rule has a string option or an object option:
> ```js
> {
>     "comma-dangle": ["error", "never"],
>     // or
>     "comma-dangle": ["error", {
>         "arrays": "never",
>         "objects": "never",
>         "imports": "never",
>         "exports": "never",
>         "functions": "never"
>     }]
> }
> ```

Allowing multiple elements can be misleading, as in #13165. Users might expect that an additional object can be used to override the string option.

We already fixed this in #13166 but reverted the change in #13564 per the semver policy, so this is a task for the next major version.

**Comment by masparasol** [comment]
This is a bit random, but I just funded this issue with a $2 bounty on Rysolv.com here: https://rysolv.com/issues/detail/0c32204b-018e-4e55-b043-0fab6079f6fd

Sorry if this comes off as spam, a friend of mine started the site and I funded the issue $2 to test it. Thought it might be fun if you want to claim it. It would require you to sign up to the site, so I understand if you don't want to spend the time, but just thought to mention it.

Thanks for working on the ES Lint project! :)
